server/src/utils/transform-validation-errors.ts

```diff
@@ -0,0 +1,18 @@
+import { PhacEntryError } from '../types';
+
+export const transformValidationErrors = (errors: string[]): PhacEntryError[] => {
+  return errors.reduce((errors, errorMessage) => {
+    const found = errorMessage.match(/^(?<id>.*):\[(?<index>\d+)\]\.(?<message>.*)$/);
+    if (!found) return;
+
+    const { index, id, message } = found.groups;
+    let error = errors.find(e => e.index === index);
```

Comment:
I think this is not so performant. I'd use a simple loop with a Map object to set those values.

Reply:
I updated to use object as a return value but it changes the final json object like below.
```
{
    "errors": {
        "0": {
            "id": "CVR-test1",
            "index": "0",
            "errors": [
                "date_of_birth is invalid",
                "arrival_date is invalid",
                "end_of_isolation should be later than arrival date",
                "home_phone is invalid"
            ]
        },
        "1": {
            "id": "CVR-test2",
            "index": "1",
            "errors": [
                "date_of_birth is invalid",
                "arrival_date is invalid",
                "end_of_isolation should be later than arrival date",
                "home_phone is invalid"
            ]
        },
        "2": {
            "id": "",
            "index": "2",
            "errors": [
                "covid_id is required",
                "arrival_date should be later than date of birth",
                "home_phone is invalid"
            ]
        }
    }
}
```